src/jit/lclvars.cpp

```diff
@@ -4732,25 +4732,25 @@ void Compiler::lvaAssignVirtualFrameOffsetsToLocals()
     // If the frame pointer is used, then we'll save FP/LR at the bottom of the stack.
     // Otherwise, we won't store FP, and we'll store LR at the top, with the other callee-save
     // registers (if any).
+
     int initialStkOffs = 0;
+    if (info.compIsVarArgs)
+    {
+        initialStkOffs = MAX_REG_ARG * REGSIZE_BYTES;
+        stkOffs -= initialStkOffs;
```

Comment:
You could add a comment that explains that for varargs we always save all of the integer register arguments, so that they are contiguous with the incoming stack arguments. That's why MAX_REG_ARGS is used here.